ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +216,36 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)]
+      {:content        (delay (.getInputStream conn))
```

Comment:
Since we're providing all of the information for the resource in one shot, I didn't want to pay the cost for getting data that may not be used. I assume the standard handler for jar urls is just doing file loading, which isn't that expensive. But since it's possible to register custom url handlers, I didn't want to make assumptions about their performance. I don't feel strongly about this though, and can remove the delays if you prefer. The delay for :directory? should be removed either way, since that's the value we know will always be used. However, I would like to keep the option open for implementors of `resource-info` to use delays.